src/SAML2/Compat/ContainerSingleton.php

```diff
@@ -45,4 +46,25 @@ public static function initSspContainer(): Container
     {
         return new Container();
     }
+
+
+    /**
+     * @param string $className
+     * @param array $qualifiedName
+     * @return void
+     */
+    public static function registerClass(string $className, array $qualifiedName): void
+    {
+        $this->registry[$qualifiedName] = $className;
```

Comment:
Is it possible to index an array with... array elements?

I think it's easier (and clearer) if we specify separately the namespace and the name of the element. Note also that the name should **not** be the qualified name, as the prefix is irrelevant in this case.

Reply:
Of course it's possible.. Anything can be used as an array index..

> I think it's easier (and clearer) if we specify separately the namespace and the name of the element
Not sure what you mean.. Add two separate entries for one class?

The `qualifiedName` is a naming mistake... Should be localName

Reply:
I mean that the method should receive 3 parameters: the fully qualified name of the class we are registering, and the namespace and name of the element it implements.

Actually, that can be simplified if we add a check which I think we need anyway: classes registered should be extending `AbstractXMLElement`. As such, the name of the class should match the name of the element it represents, and it should also have a `NS` constant with its namespace. So we could just pass the class name to `registerClass()`, check that it extends the required `AbstractXMLElement`, and then pick both pieces of information we need from it, and add them as the tuple indexing the registry.